homeassistant/components/stream/core.py

```diff
@@ -356,3 +358,53 @@ async def handle(
     ) -> web.StreamResponse:
         """Handle the stream request."""
         raise NotImplementedError()
+
+
+class KeyFrameConverter:
+    """Generate and hold the keyframe as a jpeg."""
+
+    def __init__(self) -> None:
+        """Initialize."""
+
+        # Keep import here so that we can import stream integration without installing reqs
+        # pylint: disable=import-outside-toplevel
+        from homeassistant.components.camera.img_util import TurboJPEGSingleton
+
+        self.packet: Packet = None
+        self.image: bytes | None = None
+        self._turbojpeg = TurboJPEGSingleton.instance()
+        self.lock = asyncio.Lock()
+        self._codec_context: CodecContext | None = None
+
+    def create_codec_context(self, codec_context: CodecContext) -> None:
+        """Create a codec context to be used for decoding the keyframes."""
+
+        # Keep import here so that we can import stream integration without installing reqs
+        # pylint: disable=import-outside-toplevel
+        from av import CodecContext
+
+        self._codec_context = CodecContext.create(codec_context.name, "r")
+        self._codec_context.extradata = codec_context.extradata
+        self._codec_context.skip_frame = "NONKEY"
+        self._codec_context.thread_type = "NONE"
+
+    def generate_image(self, width: int | None, height: int | None) -> bytes | None:
+        """Generate the keyframe image. This is called in an executor thread."""
+        if not (self._turbojpeg and self.packet and self._codec_context):
+            return self.image
```

Comment:
This variable does not appear to be set anywhere, or am i missing it?

Reply:
It is set by `get_image`

Reply:
Thanks.